datanode/entities/proposal.go

```diff
@@ -127,14 +138,18 @@ func (p Proposal) ToProto() *vega.Proposal {
 		errDetails = ptr.From(p.ErrorDetails)
 	}
 
+	batchID := ptr.From(p.BatchID.String())
```

Comment:
Does this need to be the below, so for a non-batch-proposal the `batchID` will be `nil` and not a pointer to an empty string? 
```suggestion
        var batchID *string
        if len(p.BatchId) != 0 {
            batchID := ptr.From(p.BatchID.String())
        }
```